src/meteodatalab/operators/regrid.py

```diff
@@ -485,11 +486,13 @@ def iconremap(
 
     gx, gy = np.meshgrid(dst.x, dst.y)
     transformer_dst = Transformer.from_crs(dst.crs.wkt, utm_crs)
-    points_dst = transformer_dst.transform(gx.flat, gy.flat)
+    points_dst = transformer_dst.transform(gy.flat, gx.flat)
 
     xy = np.array(points_src).T
     uv = np.array(points_dst).T
 
     indices, weights = _linear_weights_cropped_domain(xy, uv)
 
-    return _icon2regular(field, dst, indices, weights)
+    return _icon2regular(field, dst, indices, weights).assign_coords(
+        lon=(("y", "x"), gx), lat=(("y", "x"), gy)
```

Comment:
should this be lon or longitude?

Reply:
this only applies if the dst grid is defined in geolatlon CRS